__tests__/test-utils/mock-data/manager.js

```diff
@@ -0,0 +1,398 @@
+// @flow
+
+import clone from 'lodash/cloneDeep'
+
+import UserPermissions from '../../../lib/common/user/UserPermissions'
+import UserSubscriptions from '../../../lib/common/user/UserSubscriptions'
+
+import type {
+  Bounds,
+  FeedVersion,
+  Project,
+  SummarizedFeedVersion
+} from '../../../lib/types'
+
+/**
+ * Make a mock deployment given a project and some FeedVersions. This is a
+ * function so that circular references can be defined.
+ */
+export function makeMockDeployment (
+  project: Project,
+  feedVersions: Array<FeedVersion> = []
+) {
+  return {
+    customBuildConfig: null,
+    customRouterConfig: null,
+    dateCreated: 1553292345720,
+    deployedTo: null,
+    feedSourceId: null,
+    // need to map FeedVersion to SummarizedFeedVersion
+    feedVersions: (feedVersions.map(
+      (version: FeedVersion): SummarizedFeedVersion => (
+        {
+          feedSource: version.feedSource,
+          id: version.id,
+          nextVersionId: version.nextVersionId,
+          previousVersionId: version.previousVersionId,
+          updated: version.updated,
+          validationResult: version.validationSummary,
+          version: version.version
+        }
+      )
+    ): Array<SummarizedFeedVersion>),
+    id: 'mock-deployment-id',
+    invalidFeedSources: [],
+    lastDeployed: null,
+    lastUpdated: 1553292345726,
+    name: 'mock-deployment',
+    organizationId: null,
+    osmFileId: null,
+    otpCommit: null,
+    project,
+    projectBounds: (null: ?Bounds), // not sure why this flow type annotation is needed (see https://git.io/fjJCm)
+    routerId: (null: ?string), // not sure why this flow type annotation is needed (see https://git.io/fjJCm)
+    user: (null: ?any) // not sure why this flow type annotation is needed (see https://git.io/fjJCm)
+  }
+}
+
+// a mock project without any feeds or deployments
+export const mockProject = {
+  autoFetchFeeds: true,
+  autoFetchHour: 0,
+  autoFetchMinute: 0,
+  bounds: null,
+  buildConfig: {
+    fares: null,
+    fetchElevationUS: null,
+    stationTransfers: null,
+    subwayAccessTime: null
+  },
+  dateCreated: 1526305321824,
+  defaultTimeZone: null,
+  deployments: [],
+  feedSources: [],
+  id: 'mock-project-id',
+  lastUpdated: 1553236399556,
+  name: 'mock-project',
+  organizationId: null,
+  otpServers: [],
+  pinnedDeploymentId: null,
+  routerConfig: {
+    carDropoffTime: null,
+    numItineraries: null,
+    requestLogFile: null,
+    stairsReluctance: null,
+    updaters: null,
+    walkSpeed: null
+  },
+  useCustomOsmBounds: false,
+  user: null
+}
+
+// a mock feed with a version
+export const mockFeedWithVersion = {
```

Comment:
It seems like this should have a type defined. Is there a reason why it doesn't?

Reply:
The typing of these mock objects are checked when they are sent to various methods as function arguments. If the object structure doesn't match the argument type, a flow error occurs.